lib/tty/progressbar.rb

```diff
@@ -290,7 +290,11 @@ def update(options = {})
     #
     # @api public
     def current=(value)
-      value = [0, [value, total].min].max
+      unless value.is_a?(Numeric)
```

Comment:
I must admit I'm torn about this change. I feel this type of check is unnecessary in Ruby. Wouldn't you agree? I use it sparingly. Also, it adds new precedence for other methods. Do we then add checks for every public method like ratio, advance etc.?

Reply:
I agree. The reason I added it was that I noticed that `current=nil` previously raised an error and didn't here. I didn't explore the implications of changing `current=` to support nil and wasn't sure if it would break anything else.

I would be happy to:
* Remove this check entirely and remove the test I added about `nil`, if you think that makes sense
* Change this to be a check only for `nil`, if that is a forbidden value

Just let me know what you think, I don't have strong convictions about this check, just wanted to anticipate any possible side effects of the `.compact` that I added.

Reply:
It certainly feels like a game of whack-a-mole. I'm curious how did you stumble on this issue? I used the intermediate progress but never needed to set the current value at the same time. 

Regardless, I'm swaying towards adding a check for nil and raising an error to preserve the current behaviour. Actually, I'm warming up to the idea of checking for numeric values. Maybe we could change the error to speak about values rather than classes? Something like `expected a numeric value but got #{value.inspect} instead.`?

Reply:
> I'm curious how did you stumble on this issue? I used the intermediate progress but never needed to set the current value at the same time.

My application has an indeterminate use case (exporting data) where we can easily communicate how many records have been exported but we don't necessarily know the total number of values to export. As a result, I have been using this API, and that's how I discovered that this API only supports determinate progress bars.

> Maybe we could change the error to speak about values rather than classes?

Sure, I'll amend the commit.